lib/ffi/pointer.rb

```diff
@@ -96,12 +96,16 @@ def write_string_length(str, len)
     # @param [String] str string to write
     # @param [Numeric] len length of string to return
     # @return [self]
-    # Write +str+ in pointer's contents, or first +len+ bytes if 
-    # +len+ is not +nil+.
+    # Write +str+ in pointer's contents.
+    # If +len+ is given, write the first +len+ bytes of +str+
+    # and append a final \0 byte after the string.
     def write_string(str, len=nil)
-      len = str.bytesize unless len
-      # Write the string data without NUL termination
-      put_bytes(0, str, 0, len)
+      if len
+        put_bytes(0, str, 0, len)
+        put_bytes(len, "\0")
```

Comment:
I think we can use `put_char(len, 0)` here instead, which should be a bit more efficient.

Reply:
Done